final_project/flaskr/templates/scripts.js

```diff
@@ -357,61 +359,45 @@ BikeMap.prototype.addTempMarker = function(lat, lng, userId, address) {
 }
 
 BikeMap.prototype.createMarker = function(state) {
-    
-    if (this.allRacks.getLayer(state.marker_id)) {
-
-        this.marker = this.allRacks.getLayer(state.marker_id);
-        this.allRacks.getLayer(state.marker_id)._popup.setContent(this.popupContent(state));
-        
-        // ALSO update the marker icon
-        let newMarkerIcon = buildMarkerIcon(state.markerColor);
-        this.allRacks.getLayer(state.marker_id).setIcon(newMarkerIcon);
-        return this.allRacks.getLayer(state.marker_id);
-    }
+    let marker;
+    // if this marker already exists
+    this.allRacks.eachLayer(layer => {
+       if (layer.options.uniqueId == state.rack_id) {
+          
+           layer._popup.setContent(this.popupContent(state));
+           // Also update the marker icon
```

Comment:
You do this in both cases. Take it out of here and out of the `if` below and put it at the end of the function.

All this `eachLayer` needs to do is find the marker. That's it.

```
content = this.popupContent(state);
eachLayer(layer => { if (layer.options.uniqueid === state.rack_id) marker = layer; });
if (marker) {
  // We should use Layer.setPopupContent() here, but can't because it doesn't redraw.
  marker._popup.setContent(this.popupContent(state));
} else {
  // We didn't find a marker, make a new one.
  marker = L.marker(...);
  this.allRacks.addLayer(marker);
  if (state.status === "approved") {
    this.allApproved.addLayer(marker);
  } else {
    this.allNotApproved.addLayer(marker);
  };
  // Not showing the popup yet.
  marker.bindPopup(content);
}

Reply:
Oh, forgot to add it at the end.

```
  <blahblah create icon and stuff>
  return marker;
}
```